### PR TITLE
PYIC-1936 - added validation using hmpo to check if first and middle names do not exceed 30 characters

### PR DESCRIPTION
**Pull request by LaiTang-GDS**
### What changed
added custom validation hmpo to check if first and middle names do not exceed 30 characters

- [PYIC-1936](https://govukverify.atlassian.net/browse/PYIC-1936)

### CODE REVIEW

File: src/locales/en/fields.yml [hunk 1]
```diff
@@ -20,13 +20,15 @@ firstName:
     default: 'Enter your first name as it appears on your passport'
     maxlength: 'Enter your first name as it appears on your passport'
     regexpassport: 'Enter your first name as it appears on your passport'
+    combinedNameLength: 'Enter your first and middle names as they appear on your passport'
```
[the remark this answers is not in the document]
Comment (LaiTang-GDS) [reply src/locales/en/fields.yml]: good spot, fixed

File: src/app/passport/fields.js [hunk 2]
```diff
@@ -1,34 +1,51 @@
...
   },
   firstName: {
     type: "text",
-    validate: ["required", 
-    { type: "maxlength", arguments:[30]},
-    { type: "regexpassport", fn: value => value.match(/^[a-zA-Z .'-]*$/)}
+    validate: ["required",
+      { type: "maxlength", arguments:[30]},
+      { type: "regexpassport", fn: value => value.match(/^[a-zA-Z .'-]*$/)},
+      firstNameMiddleNameLengthValidatorObj
```
Comment (AmritSidhu) [inline src/app/passport/fields.js:39]: `{ type: "firstNameMiddleNameLength", firstNameMiddleNameLengthValidatorObj}`

File: src/app/passport/fields.js [hunk 3]
```diff
@@ -1,34 +1,51 @@
...
+  const middleNameLength = validators.string(middleName) ? middleName.length : 0;
+  const firstNameLength = validators.string(firstName) ? firstName.length : 0;
+
+  return (middleNameLength + firstNameLength) <= length;
+}
+
+const firstNameMiddleNameLengthValidatorObj = {type: 'firstNameMiddleNameLength' ,  fn:firstNameMiddleNameLengthValidator, arguments: [30, 'firstName', 'middleNames']};
+
 module.exports = {
+  firstNameMiddleNameLengthValidator: firstNameMiddleNameLengthValidator,
```
Comment (AmritSidhu) [inline src/app/passport/fields.js:16]: Whats the point of this if we are calling `firstNameMiddleNameLengthValidatorObj` as the validator?

Comment (LaiTang-GDS) [reply src/app/passport/fields.js:16]: I had to include this for the unit test to work

**Review by AmritSidhu — APPROVED** [review]

**Review by AmritSidhu — APPROVED** [review]

**Review by AmritSidhu — APPROVED** [review]

**Pull request merged**